util/codec/bytes.go

```diff
@@ -47,9 +47,12 @@ import (
 // EncodeBytes guarantees the encoded value is in ascending order for comparison,
 // The encoded value is >= SmallestNoneNilValue and < InfiniteValue.
 func EncodeBytes(b []byte, data []byte) []byte {
+	// Allocate more space to avoid unnecessary slice growing
+	bs := make([]byte, len(b), len(b)+len(data)+20)
```

Comment:
No, meta key codec use this way, and you can see the EncodeKey also uses it. 
So maybe the best is if b is nil, we alloc a proper capability.